src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -642,8 +642,29 @@ static bool _rasterLinearGradientRect(SwSurface* surface, const SwBBox& region,
         }
     //Opaque Gradient
     } else {
-        for (uint32_t y = 0; y < h; ++y) {
-            fillFetchLinear(fill, buffer + y * surface->stride, region.min.y + y, region.min.x, 0, w);
+        if (surface->compositor) {
+            uint32_t *shape = fill->ctable;
+            auto cbuffer = surface->compositor->image.data + (region.min.y * surface->stride) + region.min.x;
+            auto sbuffer = shape + (region.min.y ) + region.min.x;
+            for (uint32_t y = 0; y < h; ++y) {
+                auto dst = &buffer[y * surface->stride];
+                auto cmp = &cbuffer[y * surface->stride];
+                auto src = &sbuffer[y];
```

Comment:
I think we can locate the condition out of the for loop. What do you think? 
ex) 
if (AlphaMask)  for (x = 0; x < w; ++x, ++dst, ++cmp) ...
else if (InvAlphaMas)  for (x = 0; x < w; ++x, ++dst, ++cmp) ...

Reply:
Thanks, of course.
Changed